week2/exercise12.js

```diff
@@ -0,0 +1,241 @@
+        "id": 6,
+        "name": "charizard",
+        "height": 17,
+        "weight": 905,
+        "types": [
+            {
+                "slot": 1,
+                "type": {
+                    "name": "fire",
+                    "url": "https://pokeapi.co/api/v2/type/10/"
+                }
+            },
+            {
+                "slot": 2,
+                "type": {
+                    "name": "flying",
+                    "url": "https://pokeapi.co/api/v2/type/3/"
+                }
+            }
+        ]
+    },
+    {
+        "id": 7,
+        "name": "squirtle",
+        "height": 5,
+        "weight": 90,
+        "types": [
+            {
+                "slot": 1,
+                "type": {
+                    "name": "water",
+                    "url": "https://pokeapi.co/api/v2/type/11/"
+                }
+            }
+        ]
+    },
+    {
+        "id": 8,
+        "name": "wartortle",
+        "height": 10,
+        "weight": 225,
+        "types": [
+            {
+                "slot": 1,
+                "type": {
+                    "name": "water",
+                    "url": "https://pokeapi.co/api/v2/type/11/"
+                }
+            }
+        ]
+    },
+    {
+        "id": 9,
+        "name": "blastoise",
+        "height": 16,
+        "weight": 855,
+        "types": [
+            {
+                "slot": 1,
+                "type": {
+                    "name": "water",
+                    "url": "https://pokeapi.co/api/v2/type/11/"
+                }
+            }
+        ]
+    },
+    {
+        "id": 10,
+        "name": "caterpie",
+        "height": 3,
+        "weight": 29,
+        "types": [
+            {
+                "slot": 1,
+                "type": {
+                    "name": "bug",
+                    "url": "https://pokeapi.co/api/v2/type/7/"
+                }
+            }
+        ]
+    }
+]
+
+const pokemonDescription = [
+    {
+        "id": 1,
+        "description": "Loves to eat"
+    },
+    {
+        "id": 2,
+        "description": "Proud of its power"
+    },
+    {
+        "id": 3,
+        "description": "Sturdy body"
+    },
+    {
+        "id": 4,
+        "description": "Highly curious"
+    },
+    {
+        "id": 5,
+        "description": "Strong willed"
+    },
+    {
+        "id": 6,
+        "description": "Likes to run"
+    },
+    {
+        "id": 7,
+        "description": "Takes plenty of siestas"
+    },
+    {
+        "id": 8,
+        "description": "Likes to thrash about"
+    },
+    {
+        "id": 9,
+        "description": "Capable of taking hits"
+    },
+    {
+        "id": 10,
+        "description": "Mischievous"
+    }
+]
+
+function fn(pokemons, pokemonDescription)
+{
+    let list = []
+    for(let i = 0; i < pokemons.length; i++)
+    {
+        let obj = {}
+        obj.id = pokemons[i].id
+        obj.name = pokemons[i].name
+        obj.description = pokemonDescription[i].description
```

Comment:
Issue: This works only if the order of `pokemons` and `pokemonDescription` have the same order. For example if pokemon is
```
[{
        "id": 1,
        "name": "bulbasaur",
 },
 {
        "id": 2,
        "name": "charizard",
 }]
```
and pokemonDescription is
```
[{
        "id": 2,
        "description": "charizard description",
 },
 {
        "id": 1,
        "description": "bulbasaur description",
 }]
```
Then the above function will provide incorrect result.